scripts/vtk_convert.py

```diff
@@ -0,0 +1,91 @@
+from vtk import vtkPolyDataReader, vtkStructuredPointsReader, \
+    vtkXMLImageDataWriter, vtkXMLPolyDataWriter
+from girder_client import GirderClient
+import os
+import click
+import re
+
+@click.command()
+@click.option('--inputfolderid', required=True, prompt=True, help='Source Folder ID')
+@click.option('--outputfolderid', required=True, prompt=True, help='Destination Folder ID')
+@click.option('--apikey', required=True, prompt=True, help='API Key from Girder')
+@click.option('--hostname', default='https://data.computational-biology.org/api/v1/', prompt=True, help='URL, make sure to include /api/v1/ at the end')
+def main(inputfolderid, outputfolderid, apikey, hostname):
+  gc = GirderClient(apiUrl=hostname)
+  gc.authenticate(apiKey=apikey)
+
+  def getFiles(itemData):
+    print('getting ' + itemData['name'])
+    dataFiles = gc.get('item/' + itemData['_id'] + '/files')
+    return dataFiles[0]
+
+  def getItems(folderData):
+    folderInfo = gc.get('folder/' + folderData['_id'] + '/details')
+    folderItems = gc.get('item',
+      parameters={
+        'folderId': folderData['_id'],
+        'limit': folderInfo['nItems']
+      })
+    return map(getFiles, folderItems)
+
+  gmsFolders = gc.get('folder',
+    parameters={
+      'parentType': 'folder',
+      'parentId': inputfolderid,
+    })
+
+  gmsFileNames = map(getItems, gmsFolders)
+
+  for row in gmsFileNames:
+    for elem in row:
+      oldName = elem['name']
+      print('dowloading ' + oldName)
+      try:
+        gc.downloadFile(elem['_id'], os.path.abspath(oldName))
+
+        reader = vtkPolyDataReader()
+        writer = vtkXMLPolyDataWriter()
+        fileType = ''
+        if re.search("^geometry_", oldName):
+          reader = vtkStructuredPointsReader()
+          reader.ReadAllScalarsOn()
+          writer = vtkXMLImageDataWriter()
+          fileType = 'geometry_'
+        elif re.search("^spore_", oldName):
+          fileType = 'spore_'
+        elif re.search("^macrophage_", oldName):
+          fileType = 'macrophage_'
+        else:
+          raise Exception('Invalid File: {}'.format(oldName))
+
+        reader.SetFileName(oldName)
+        reader.Update()
+        data = reader.GetOutput()
+
+        end = oldName[len(fileType):]
```

Comment:
This whole block is pretty complicated. I think what you want is to write out a number into a string with a fixed number of digit places and leading zeros as necessary. See https://pyformat.info/#number_padding for the syntax, which you can also use with f-strings: https://stackoverflow.com/questions/45310254/fixed-digits-after-decimal-with-f-strings

Reply:
So my computer currently has python 2.7 and I think the number padding is for python 3. Should I just update my python to be python 3?